From 1cf4bb906924b6e3fcbd28e14ff0ab975e6f676a Mon Sep 17 00:00:00 2001
From: Yufei Gu <yufei@apache.org>
Date: Mon, 28 Aug 2017 10:09:46 -0700
Subject: [PATCH 2624/2848] YARN-7099.
 ResourceHandlerModule.parseConfiguredCGroupPath
 only works for privileged yarn users.
 (Contributed by Miklos Szegedi via Yufei Gu)

Conflict:
ResourceHandlerModule.java: removed non-exising comment.

(cherry picked from commit d5bc9582793de6ab2d2e489cde293bf076229a1c)

(cherry picked from commit 221c1bd)

Change-Id: I70e43e10d79b85e30ca5027e3d55586dea81a8c0
---
 .../linux/resources/ResourceHandlerModule.java     |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/src/main/java/org/apache/hadoop/yarn/server/nodemanager/containermanager/linux/resources/ResourceHandlerModule.java b/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/src/main/java/org/apache/hadoop/yarn/server/nodemanager/containermanager/linux/resources/ResourceHandlerModule.java
index 6314d8a..10214fb 100644
--- a/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/src/main/java/org/apache/hadoop/yarn/server/nodemanager/containermanager/linux/resources/ResourceHandlerModule.java
+++ b/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/src/main/java/org/apache/hadoop/yarn/server/nodemanager/containermanager/linux/resources/ResourceHandlerModule.java
@@ -194,7 +194,7 @@ static void nullifyResourceHandlerChain() throws ResourceHandlerException {
       // Collect the valid subsystem names
       cgroupList.retainAll(validCGroups);
       if (!cgroupList.isEmpty()) {
-        if (candidate.isDirectory() && candidate.canWrite()) {
+        if (candidate.isDirectory()) {
           pathSubsystemMappings.put(candidate.getAbsolutePath(), cgroupList);
         }
       }
-- 
1.7.9.5

